### ISSUE TITLE
Define a property to identify the cross-targeting/SDK project type

### ISSUE THREAD

**Issue by AArnott** [dotnet/sdk#534]
The cross-targeting / SDK project type offer new functionality that a build extension may want to detect in order to reliably leverage (as opposed to setting properties and blindly hoping the new functionality is present).

Please define a new property that a build extension can check for to detect the use of the new SDK attribute in the project file.

Per a brief side discussion on #528 with @nguerrera.

**Comment by onovotny** [comment]
More than just a property, we need to know how to call it when referring to it in documentation. It needs a "marketing name" too.

**Comment by dsplaisted** [comment]
I'd suggest defining a property that indicates the version number of the .NET SDK that is being used.  That way if a NuGet package depends on functionality in a new version of an SDK, the package can have MSBuild logic that will generate an error saying you need to use a newer SDK if the current version isn't high enough.

**Comment by nguerrera** [comment]
@dsplaisted Makes sense. Which version should we use? I think it should be System.Version parse-able to be friendly to msbuild's support for semantically correct comparisons of those.

**Comment by dsplaisted** [comment]
@nguerrera yes, I think it should be a System.Version version, and it would be nice to have another property indicating the semantic version suffix (ie "alpha-1234" or whatever)

**Comment by ericstj** [comment]
+1 to this.  I'm actually blocked on this.  I'm creating a nuget package that needs to understand if its in nuget2 vs nuget3 vs nuget4 mode.  I thought I had a good condition for this: https://github.com/dotnet/standard/commit/09e9f9ecda4aa9eeedc2aadbd1b0bdaff8d6e010#diff-de762529dcf068a5867280f815198927R9 but it looks like the properties I was using are actually set by targets imported *after* those imported by NuGet packages so my condition doesn't work 😕.  Can you please set a property in `.props` that indicates the project is in PackageReference mode?

**Comment by ericstj** [comment]
+1 again.  I'd like to poison our new framework packages and point folks to update their SDK.  I'd really like something explicit to trigger that on, rather than a side-effect of implementation.

**Comment by AArnott** [comment]
Maybe if folks +1 the top-level issue itself it will gain more visibility. :)

**Comment by nguerrera** [comment]
@dsplaisted can you poach this as part of your bug push?

**Comment by ericstj** [comment]
Ideally this is a property to identify both the SDK project type and the SDK version.

**Comment by nguerrera** [comment]
What do you mean by sdk project type?

**Comment by ericstj** [comment]
What @AArnott described initially.

**Comment by nguerrera** [comment]
So "is this an sdk project or not" not "what kind of sdk project is this". Got it.

**Comment by nguerrera** [comment]
Implementing the boolean part in #1242. Plumbing the version through is more problematic so keeping that separate.

**Comment by livarcocc** [comment]
Closing this issue in the SDK side and moving it to the CLI side to provide an SDK version property. We should do something similar to Microsoft.NETCoreSdk.BundledVersions.props.

**Comment by livarcocc** [comment]
This issue was moved to dotnet/cli#7051
